lib/elixir/lib/calendar/date.ex

```diff
@@ -149,6 +159,11 @@ defmodule Date do
     range(first, first_days, last, last_days, calendar, step)
   end
 
+  def range(%{calendar: calendar} = first, %Duration{} = duration, step) do
```

Comment:
We should probably guard this too to check that step is a non-zero integer and update the ArgumentError clause & message?